.build/docker/almalinux-build.docker

```diff
@@ -0,0 +1,60 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM almalinux:8
+MAINTAINER Apache Cassandra <dev@cassandra.apache.org>
+
+# CONTEXT is expected to be cassandra/.build
+
+ENV BUILD_HOME=/home/build
+ENV RPM_BUILD_DIR=$BUILD_HOME/rpmbuild
+ENV DIST_DIR=/dist
+ENV CASSANDRA_DIR=$BUILD_HOME/cassandra
+ARG UID_ARG=1000
+ARG GID_ARG=1000
+
+LABEL org.cassandra.buildenv=almalinux
+
+VOLUME ${DIST_DIR}
+VOLUME ${CASSANDRA_DIR}
+
+RUN echo "Building with arguments:" \
+    && echo " - DEB_DIST_DIR=${DEB_DIST_DIR}" \
+    && echo " - BUILD_HOME=${BUILD_HOME}" \
+    && echo " - CASSANDRA_DIR=${CASSANDRA_DIR}" \
+    && echo " - UID_ARG=${UID_ARG}" \
+    && echo " - GID_ARG=${GID_ARG}"
+
+# install deps
+RUN yum -y install \
+   ant \
+   git \
+   java-1.8.0-openjdk-devel \
+   java-11-openjdk-devel \
```

Comment:
Will these packages drift in their version? i.e. as new JDK 8, 11, and 17 are released, will this potentially change underneath us in our build scripts?

Might be worth having these versions also declared in a central .sh file we source here. One-stop-shop for all env var for the build we can then update as needed.

There may be an issue where different environments have different JDK releases; in that case we could declare the desired JDK version locally in their respective docker files and reference those variables.

Reply:
Yes, that's why they're in-tree, and why the image tags and container names include these variables.

Reply:
So to make sure I understand, the image tags / container names include variables that'll constrain `java-1.8.0-openjdk-devel` to a singular reproducible version on that env?

Reply:
image names are like: `apache/cassandra-ubuntu2004_test:b09545573ea248c487eba7547094821d` 
 where the tag is a md5sum of the dockerfile.

since each each supports the different axis values.

container names are like contain the values, plus a random hex, so they are easy to identifer.
ref:
 - https://github.com/apache/cassandra/pull/2351/files#diff-e8f0cb5844d900a3d89a43743814930e66575abc2b6731c07d6a691de51a4d65R97
 -  https://github.com/apache/cassandra/pull/2351/files#diff-d6af0f7ab7ddf83429e201fad8288dbdfb90fbd89fd9cf1bbb74fd06a65eb0d8R189

(i will make those lines easier to read…)

The idea here is that the tag (md5sum) allows us to
- deploy the image to dockerhub, allowing it be be downloaded instead of built,
- be easily cached between builds,
- provide reproducible builds (answers your concern most IIUC)
- be able to patch the dockerfile and have it used in the CI run without fuss